libs/translator/writer/write_shader.cpp

```diff
@@ -82,4 +82,40 @@ void UsdArnoldWriteShader::Write(const AtNode *node, UsdArnoldWriter &writer)
     }
     // Ensure all shaders have an output attribute
     prim.CreateAttribute(str::t_outputs_out, SdfValueTypeNames->Token, false);
+
+    if (!(writer.GetMask() & AI_NODE_SHAPE)) {
+        // If shapes are not exported and a material is specified for this shader, let's create it
+        AtString materialName;
+        bool isDisplacement = false;
+        // material_surface / material_displacement / material_volume are user data 
+        // authored by the arnold plugins when a shader library is exported, so that 
+        // materials can be restored at import. We can use this to create the
+        // USD material primitives #2047
+        if (AiNodeLookUpUserParameter(node, str::material_surface)) {
+            materialName = AiNodeGetStr(node, str::material_surface);
+        } else if (AiNodeLookUpUserParameter(node, str::material_displacement)) {
+            materialName = AiNodeGetStr(node, str::material_displacement);
+            isDisplacement = true;
+        } else if (AiNodeLookUpUserParameter(node, str::material_volume)) {
+            // Note that volume assignments are treated the same way as
+            // surface shader assignments in our usd support
+            materialName = AiNodeGetStr(node, str::material_volume);
+        }
+
+        if (!materialName.empty()) {
+            std::string matName(materialName.c_str());
+            _SanitizePrimName(matName);
+            std::string mtlScope = writer.GetScope() + writer.GetMtlScope();
+            writer.CreateScopeHierarchy(SdfPath(mtlScope));
+            matName = mtlScope + matName;
+            UsdShadeMaterial mat = UsdShadeMaterial::Define(writer.GetUsdStage(), SdfPath(matName));
+            const TfToken arnoldContext("arnold");
+            std::string shaderOutput = prim.GetPath().GetString() + std::string(".outputs:out");
+            UsdShadeOutput matOutput = (isDisplacement) ? 
```

Comment:
here we only use the displacement and surface as the volume are treated the same way as surface. In the api there is a [CreateVolumeOutput](https://openusd.org/release/api/class_usd_shade_material.html#a4abdf5ff051e043d5b4b8bc5dd0a70f1) function that we could also use, that would handled internal changes by the api. What do you think ?